File: tests/openmp_examples/devices/target_mapper.2.c
#pragma omp declare mapper( top_id: dzmat_t v) map(v.r_m[0:100/2][0:100], v.i_m[0:100/2][0:100] )
#pragma omp declare mapper(bottom_id: dzmat_t v) map(v.r_m[100/2:100/2][0:100], v.i_m[100/2:100/2][0:100] )
#pragma omp target map(mapper(top_id), tofrom: a,b) device(0) firstprivate(is,ie) nowait
#pragma omp target map(mapper(bottom_id), tofrom: a,b) device(1) firstprivate(is,ie) nowait
#pragma omp taskwait

File: tests/openmp_examples/devices/target_mapper.2.f90
!$omp     declare mapper( left_id: dzmat_t :: v) map( v%r_m(N,  1:N/2),v%i_m(N, 1:N/2))
!$omp     declare mapper(right_id: dzmat_t :: v) map( v%r_m(N,N/2+1:N),v%i_m(N,N/2+1:N))
!$omp   target map(mapper( left_id), tofrom: a,b) device(0) firstprivate(is,ie) nowait
!$omp   end target
!$omp   target map(mapper(right_id), tofrom: a,b) device(1) firstprivate(is,ie) nowait
!$omp   end target
!$omp   taskwait
